File: java/src/main/java/io/hatake/core/SectionDefinition.java
package io.hatake.core;

import java.util.List;
import java.util.Map;

/**
 * フォームの1区画。
 *
 * <p>{@code visibleWhen} があると区画ごと出し分ける（「法人のときだけ請求先の枠を出す」）。
 * 隠れている区画の項目は {@link FormValidator} も検証しない＝入力できないものは求めない。
 */
public record SectionDefinition(
        String title,
        List<FieldDefinition> fields,
        Map<String, Object> visibleWhen) {

    /** 条件を持たない普通の区画。 */
    public SectionDefinition(String title, List<FieldDefinition> fields) {
        this(title, fields, null);
    }
}

File: flutter/packages/hatake_core/lib/src/definition/section_definition.dart
import 'package:equatable/equatable.dart';

import 'field_definition.dart';
import 'layout_definition.dart';

/// A titled group of fields within a form.
class SectionDefinition extends Equatable {
  /// Optional section heading.
  final String? title;

  final List<FieldDefinition> fields;

  final LayoutDefinition layout;

  /// Show the whole section only when the condition matches the current record
  /// (see `evaluateCondition`). Null = always visible.
  ///
  /// A hidden section's fields are not validated either — the same rule as a
  /// field hidden by its own `visibleWhen`.
  final Map<String, Object?>? visibleWhen;

  const SectionDefinition({
    this.title,
    this.fields = const [],
    this.layout = LayoutDefinition.single,
    this.visibleWhen,
  });

  @override
  List<Object?> get props => [title, fields, layout, visibleWhen];
}
